### PR TITLE
chore: Upgrade kind to v0.23.0; use k8s 1.30.0 for integration tests

### PR DESCRIPTION
**Pull request by maxsmythe**
**What this PR does / why we need it**:

We need k8s 1.30.0 so we can test constraints/templates against the newest apparmor behavior, where the apparmor profile is described in the securityContext as opposed to in an annotation.

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: library/pod-security-policy/proc-mount/samples/psp-proc-mount/example_allowed.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 apiVersion: v1
 kind: Pod
 metadata:
-  name: nginx-proc-mount-disallowed
```
Comment (ritazh) [inline library/pod-security-policy/proc-mount/samples/psp-proc-mount/example_allowed.yaml:4]: I think you need to run 'make generate generate-website-docs generate-artifacthub-artifacts' to re-generate the templates and docs for this change

Comment (maxsmythe) [reply library/pod-security-policy/proc-mount/samples/psp-proc-mount/example_allowed.yaml:4]: Thanks! I thought I ran `make generate-all` I guess not :/

Hopefully this fixes the tests

Comment (maxsmythe) [reply library/pod-security-policy/proc-mount/samples/psp-proc-mount/example_allowed.yaml:4]: Tests are still failing

@sozercan @ritazh @JaydipGabani Any idea why the test workflow is not picking up my changes?

Here is the error:

```
# actual: The Pod "nginx-proc-mount-disallowed" is invalid: spec.containers[0].securityContext.procMount: Invalid value: "Unmasked": `hostUsers` must be false to use `Unmasked`
```

This should not be encountered given the changes made

Comment (maxsmythe) [reply library/pod-security-policy/proc-mount/samples/psp-proc-mount/example_allowed.yaml:4]: Made tests more verbose. Looks like the changes were taking effect.

Digging through the k8s code, it looks like UserNamespacesSupport feature gate needs to be enabled for `hostUsers` config to be honored.

**Review by ritazh — APPROVED** [review]
LGTM

**Pull request merged**